projects/storefrontapp/src/app/qualtrics-extended.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import {
+  CartService,
+  QualtricsConfig,
+  QualtricsConfigService,
+  WindowRef,
+} from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Injectable({ providedIn: 'root' })
+export class QualtricsExtendedService extends QualtricsConfigService {
+  constructor(
+    winRef: WindowRef,
+    config: QualtricsConfig,
+    private cartService: CartService
+  ) {
+    super(winRef, config);
+  }
+
+  isDataLoaded(): Observable<boolean> {
```

Comment:
and some comment that helps understanding we're waiting for cart data before Qualtrics kicks in